### DISCUSSION TITLE
Using PyQt to Output JSON

### DISCUSSION THREAD

**Discussion by jmdelahanty** [wehr-lab/autopilot#171] [Q&A] [answered]
Hello Autopilots!

I was curious about how to best output json files from a GUI. Is there a good spot in the repo here or resources out there you like for making a GUI that outputs json in PyQt5/6? I can't seem to get the callbacks to work right yet...

**Comment by sneakers-the-rat** [comment, marked as answer]
Not quite sure what you have in mind, and don't use PyQt, but with PySide a minimal example might look something like this:

```python
from PySide2 import QtWidgets
import json

class MyWidget(QtWidgets.QWidget):
    def __init__(self):
        self.layout = QtWidgets.QVBoxLayout()
        self.setLayout(self.layout)

        self.input = QtWidgets.QLineEdit()
        self.save_button = QtWidgets.QPushButton()
        self.save_button.clicked.connect(self.save)

        self.layout.addWidget(self.input)
        self.layout.addWidget(self.save_button)

    def save(self):
        my_text = self.input.text()
        with open('output_file.json', 'w') as jfile:
            json.dump(my_test, jfile)

```

**Reply by jmdelahanty** [reply]
Thank you! This is a much nicer example than a lot of the ones I've found so far online.

The goal is to make a GUI that people can use for inputting surgery metadata/other metadata into json files I use for generating NWB compliant datasets. One day it would also be really cool to have a GUI for the project that runs one of our microscopes, but that seems a lot harder to accomplish so far. GUIs are tough!

**Reply by sneakers-the-rat** [reply]
oh i just made something for that: https://github.com/wehr-lab/autopilot/blob/detour-datamodel/autopilot/gui/widgets/model.py

**Reply by jmdelahanty** [reply]
Wow very cool! I'll see if I can get this running properly and show it to my labmates one day. Thanks Jonny! You always seem to have the solution haha
